apps/ui/src/views/Space/Treasury.vue

```diff
@@ -1,12 +1,20 @@
 <script setup lang="ts">
-import { shorten } from '@/helpers/utils';
+import { shorten, toKebabCase } from '@/helpers/utils';
 import { RequiredProperty, Space, SpaceMetadataTreasury } from '@/types';
 
 const props = defineProps<{ space: Space }>();
 
 const { setTitle } = useTitle();
+const route = useRoute();
+
+const activeTreasuryId = computed(() => {
+  if (!route.params.name) return 0;
+  const index = filteredTreasuries.value.findIndex(
+    t => toKebabCase(t.name) === (route.params.name as string)
+  );
+  return index;
+});
```

Comment:
Overall I think it looks good, but using just id might be better there: https://github.com/snapshot-labs/workflow/issues/193#issuecomment-2337502921

Solves bunch of issues, won't be canonical though (if at some point treasuries change order).

Reply:
Might not be compatible with my ideas here: https://github.com/snapshot-labs/sx-monorepo/issues/728

It may be an edge case, but when you have multiple treasuries/delegations, that users have added at different dates, they could feel that some added later (so more recent) are more important, and should be shown first.

When you have a few treasuries/delegations, and they all don't have a short name, you don't want your main treasuries/delegations to be hidden by the horizontal scroller